pkg/query/config.go

```diff
@@ -0,0 +1,53 @@
+package query
+
+import (
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/httpconfig"
+	"gopkg.in/yaml.v2"
+)
+
+type EndpointConfig struct {
+	Endpoints   []string                  `yaml:"addresses"`
+	EndpointsSD []httpconfig.FileSDConfig `yaml:"file_sd_configs"`
+}
+
+// LoadConfig returns list of per-endpoint TLS config.
+func LoadConfig(confYAML []byte, endpointAddrs []string, globalFileSDConfig httpconfig.FileSDConfig) ([]EndpointConfig, error) {
+	var endpointConfig []EndpointConfig
+
+	if len(confYAML) > 0 {
+		if err := yaml.UnmarshalStrict(confYAML, &endpointConfig); err != nil {
+			return nil, err
+		}
+
+	}
+
+	// Adding --store, rule, metadata, target, exemplar and --store.sd-files, if provided.
+	// Global TLS config applies until deprecated.
+	if len(endpointAddrs) > 0 || globalFileSDConfig.Files != nil {
+		cfg := EndpointConfig{}
+		cfg.Endpoints = endpointAddrs
+		if globalFileSDConfig.Files != nil {
+			cfg.EndpointsSD = []httpconfig.FileSDConfig{
```

Comment:
So we can maybe add a new field to the `EndpointConfig` struct, something like `EndpointSDDiscoverer` and in `LoadConfig` we can use `file.NewDiscovery` to process all the file SD configs and populate the new field. We would then be able to use the discoverers directly in `query.go` with `fileSDCache`. Wdyt?